### ISSUE TITLE
Proposal: drop support for PHP 5.6 and move to at least 7.2

### ISSUE THREAD

**Issue by madman-81** [Torann/json-ld#55]
Since PHP support for 5.6 has stopped long ago and even version 7.1 only has security support (until December 1st), I would like to propose to set the required PHP version to >=7.2. This would, allow for some clean up of code, enable more type hinting, etc.

Please let me know if other people would like move forward too or if there are any reasons why not to do this.

**Comment by madman-81** [comment]
@Torann Would be nice to have your input on this. If you agree, I could also add a pull request for this if you like to.

**Comment by Torann** [comment]
Sounds great to me

**Comment by madman-81** [comment]
Great, I'll start working on that soon then! Since this could result in quite some changes, it would be convenient if the other pull requests are reviewed and merged before starting on this one. This makes things a bit easier with less merge conflicts. Could you have a look at them?

**Comment by Torann** [comment]
I'll look into doing this for the next release and make it the official v1.0 release

**Comment by Torann** [comment]
Just started [1.0.0](https://github.com/Torann/json-ld/tree/1.0.0) 🎆
